pkg/components/validator.go

```diff
@@ -511,17 +511,30 @@ type OCIPluginConfig struct {
 	Validator        *oci.OciValidatorSpec  `yaml:"validator"`
 }
 
-// BasicAuths returns a slice of basic authentication details for each secret.
+// BasicAuths returns a slice of basic authentication details for each rule.
 func (c *OCIPluginConfig) BasicAuths() map[string][]string {
-	auths := make(map[string][]string, len(c.Secrets))
-	for _, s := range c.Secrets {
-		s := s
-		if s.BasicAuth != nil {
-			auths[s.Name] = []string{s.BasicAuth.Username, s.BasicAuth.Password}
-		} else {
-			auths[s.Name] = []string{"", ""}
+	auths := make(map[string][]string, 0)
+
+	for _, r := range c.Validator.OciRegistryRules {
+		if r.Auth.SecretName != nil {
+			for _, s := range c.Secrets {
+				if s.Name != *r.Auth.SecretName {
+					continue
+				}
+
+				if s.BasicAuth != nil {
+					auths[r.Name()] = []string{s.BasicAuth.Username, s.BasicAuth.Password}
+				}
+			}
+			continue
+		}
+
+		if r.Auth.Basic != nil {
```

Comment:
You're reading ECR creds below but not handling them here.

Reply:
handling the ecr creds needs to be left to the plugin. The plugin will invoke each rule in sequence and there may be multiple different aws env vars to set. So handling them here is basically pointless imo